### ISSUE TITLE
[C++11] methods with ref-qualifiers fail to parse

### ISSUE THREAD

**Issue by mdouze** [swig/swig#1059]
When running `swig -module tt -c++ -python tt.h` on the following code: 

```
%inline %{

class Host : private apache::thrift::detail::st::ComparisonOperators<Host> {
 public:
    std::string host;

    const std::string& get_host() const& {
        return host;
    }
 };

%}
```

the parser fails. This is annoying because the C++ code itself is (vastly simplified) code generated by the Thrift RPC engine. When removing the `&` after `const` it compiles fine. Tested on swig 3.0.10

**Comment by ojwb** [comment]
Looks like a missing C++11 feature (ref-qualifiers):

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2007/n2439.htm

**Comment by wsfulton** [comment]
I've got a fix for this, which I'll push to master within a few days.

**Comment by wsfulton** [comment]
The full set of fixes are in 4d2d8dd80e073afbc89c8135916a28a2ce76932e.
